### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: destructuring-assignment

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1834]
This issue describes how to implement the `destructuring-assignment` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Know where destructuring can be used (in `loop` and `destructuring-bind`)
- Be able to perform simple destructuring within `loop` (such as looping over dotted pairs)
- Know how to use `destructuring-bind` to break up nested data (lists of lists and the like)
- Understand the macro-lambda list options available within `destructuring-bind` (`&rest`, `&optional`, `&key`, `allow-other-keys`, and `&whole`)
- Understand some of the limitations of destructuring in Common Lisp (limited to cons-cells)

## Out of scope

- Other features of `loop` unrelated to destructuring
- The obscure macro-lambda options (`&environment`, `&aux`, etc)

## Concepts

- `destructuring-assignment`

## Prerequisites

- `basic-lists`
- `symbols`
- `function-definition`
- `loop.basic`
- `variables`

### Hints

- [Little Bits of Lisp](https://www.youtube.com/watch?v=md9iv71W7o8)
- [Practical Common Lisp](http://www.gigamonkeys.com/book/beyond-lists-other-uses-for-cons-cells.html)
- [CL Cookbook](https://lispcookbook.github.io/cl-cookbook/data-structures.html#destructuring-bind-parameter-list)
- [Loop Destructuring](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node252.html)

### After

- [Pattern Matching With Trivia](https://lispcookbook.github.io/cl-cookbook/pattern_matching.html)

## Representer

Nothing yet

## Analyzer

We need to check that `destructuring-bind` is being used and not individual accessors like `first` or `caddr`

## Implementing

It might be interesting to show a comparison between a bit of code doing nested accesses without destructuring next to a `destructuring-bind`. Hopefully the destructuring version is convincingly clearer and shorter so that the student really understands the benefit it brings.

The `loop` destructuring is arguably more "obscure", but it would be excellent to show how you can loop over pairs or even lists representing something like a 3D point in space.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
